### ISSUE TITLE
Add CLI support for input types

### ISSUE THREAD

**Issue by dmehra** [juttle/juttle#32]
Currently the CLI ignores the input type in a program, and is therefore only useful for strings (input type: text) or numbers (which only "work" because of a hack).

To support things like moments or filter expressions we could change the CLI to do the same two-phase program execution that explorer used to do, i.e. first compile the program to get the types and options for all inputs, then execute some logic to parse the input values from the command line based on their types, and then run the program.

When this is done, add a docs example for input filter:

```
juttle --input grep='wlan0' docs/examples/inputs/filter_logs.juttle
```
filter_logs.juttle:
```
input grep: filter
-label 'Filter logs on expression:'
-description 'Sample filter expression: wlan0'
;
demo -demo 'event' grep
```

**Comment by dmehra** [comment]
@demmer please confirm that we still want this in the CLI, as opposed to outrigger.

**Comment by dmehra** [comment]
This connects to https://github.com/juttle/juttle/issues/50

**Comment by demmer** [comment]
> @demmer please confirm that we still want this in the CLI, as opposed to outrigger.

Yes -- in outrigger we have the infrastructure to properly support inputs since all juttles go through a two-phase execution in order to first extract the inputs and then instantiate specific inputs for the given type before executing the program. This way the given input type (text, number, filter, etc) will parse its input into the right expression type.
